2.13/tests/test/test.js

```diff
@@ -655,3 +723,71 @@ function validateJsonSchema (body, schema) {
   }
   return ''
 }
+
+function validateCatalogSchema (tempBody, schemaType, action) {
+  preparedRequest()
+    .get('/v2/catalog')
+    .set('X-Broker-API-Version', apiVersion)
+    .auth(config.user, config.password)
+    .expect(200)
+    .expect('Content-Type', /json/)
+    .end(function (err, res) {
+      if (err) {
+        var message = 'Get catalog result failed: ' + err
+        return message
+      }
+      var catalog = _.clone(res.body)
+      var schemaCheckResults = parametersSchemaCheck(catalog, tempBody.service_id, tempBody.plan_id, schemaType, action, tempBody.parameters)
+      if (schemaCheckResults !== '') {
+        message = 'Validate schema parameters failed!'
+        return message
+      }
+    })
+  return ''
+}
+
+function findWhichContains (obj, key, value) {
+  if (!obj) {
+    return null
+  }
+  if (obj[key] === value) {
+    return obj
+  }
+  if (Array.isArray(obj)) {
+    for (var i in obj) {
+      var found = findWhichContains(obj[i], key, value)
+      if (found) return found
+    }
+  } else if (typeof obj === 'object') {
+    for (var p in obj) {
+      if (p === key && obj[p] === value) {
+        return obj
+      }
+      found = findWhichContains(p, key, value)

```

Comment:
Should be `findWhichContains(obj[p], key, value)`. Also, you may want to rename `p` to `k` to indicate it is a "key".

Reply:
Got it, I think we should also update it in mock server.